### PR TITLE
fmod Error handling

### PR DESCRIPTION
**Pull request by Darksun2010**
I found following words in C Reference:
If y is ±0 and x is not NaN, NaN is returned and FE_INVALID is raised
If y is ±∞ and x is finite, x is returned.

So I edited fmod.c in order to make it closer to the standard.

### CODE REVIEW

File: src/fmod.c [hunk 1]
```diff
@@ -1,8 +1,10 @@
 #include <math.h>
 
 double fmod(double x, double div) {
+	if(div != div)
```
Comment (Ivoah) [inline src/fmod.c]: When would this ever be true?

**Comment by SirCmpwn** [comment]
Interesting to think of how this will work in the future in terms of signals, which don't work the Unixy way on KnightOS. Will merge when @Ivoah's comment is addressed - and also, check the [KnightOS style guide](http://www.knightos.org/contributing/guidelines.html).

File: src/fmod.c (hunk 1 above)
Comment (Darksun2010) [inline src/fmod.c]: when div is a NaN,it will be true.

Comment (SirCmpwn) [inline src/fmod.c]: Sounds good to me. Just the style issues need fixing, then.

**Comment by Darksun2010** [comment]
I think problems about style is already fixed.Sorry that I didn't read the guide carefully.

File: src/fmod.c [hunk 2]
```diff
@@ -1,8 +1,11 @@
 #include <math.h>
 
 double fmod(double x, double div) {
+	if(div != div) {
```
Comment (SirCmpwn) [inline src/fmod.c]: Add a space between if and (

**Comment by SirCmpwn** [comment]
Almost! Just one minor error.

**Comment by Darksun2010** [comment]
Now that should be OK.
But I found there was not space between "if" and "(" in origin fmod.c:P

**Comment by SirCmpwn** [comment]
Haha, no worries. Thanks!

File: src/fmod.c [hunk 3]
```diff
@@ -1,10 +1,14 @@
...
 double fmod(double x, double div) {
-	if(div == 0) {
-		return 0;
+	//if div == nan return x directly
+	if (div != div) {
+		return x;
 	}
-	while(x > div) {
+	if (div == 0) {
+		return 0.0 / 0.0;
```
Comment (aouelete) [inline src/fmod.c]: Is there some other way to return nan here? Compiling this gives a division by zero error.

Comment (SirCmpwn) [inline src/fmod.c]: Whoops. I just pushed a fix.

**Pull request merged**